gatekeeper.js

```diff
@@ -0,0 +1,28 @@
+var normalizeVersion = require('./normalize-version');
+
+var currentVersion = normalizeVersion(process.version);
+var minVersion = normalizeVersion(require('./package.json').engines.node);
+
+if (currentVersion[0] < minVersion[0]) abort();
```

Comment:
Did you consider using the `semver` package for this logic?

Reply:
🤔 now that you mention it.
I'll look into that 🙂

Reply:
After looking into it, this is my thinking is as follows.

`gatekeeper.js` needs the functionality and should be as light-weight as possible in order to abort or continue as soon as possible. Consequently, gatekeeper should not be burdened with the whole semver package, and is better off with the custom function.

Since the function will exist anyway, based on that, we may as well use it in `tools.js` because, it's still a wee bit faster and fully sufficient. Of course, if we needed even a little bit more functionality, I'd rather use `semver` in `tools.js` than expand the custom function.

However, these advantages are probably not noticeable to humans, so I wouldn't fight it if you want to use `semver` instead.

Reply:
> However, these advantages are probably not noticeable to humans

Because of this, yes I think I would prefer using `semver` instead so that we have less code to maintain and the code is more readable. Not a huge preference though and you've already done a ton of work here so if you'd rather ship it as is and defer improvements I'm fine with that too!

Reply:
Valid argument, and after thinking about it, I do agree. Done 👍